libs/jsonschema/instance.go

```diff
@@ -74,12 +73,20 @@ func (s *Schema) validateAdditionalProperties(instance map[string]any) error {
 	return nil
 }
 
+type RequiredPropertyMissingError struct {
+	Name string
+}
+
+func (err RequiredPropertyMissingError) Error() string {
+	return fmt.Sprintf("no value provided for required property %s", err.Name)
+}
+
 // This function validates that all require properties in the schema have values
 // in the instance.
 func (s *Schema) validateRequired(instance map[string]any) error {
 	for _, name := range s.Required {
 		if _, ok := instance[name]; !ok {
-			return fmt.Errorf("no value provided for required property %s", name)
+			return RequiredPropertyMissingError{Name: name}
```

Comment:
This change turns this error into its own type but the other validation functions still return `fmt.Errorf`. Could you include error types for the other cases as well?

Reply:
Removed this error type. It's no longer needed if we strictly stick to the JSON schema.